torch/lib/c10d/reducer.cpp

```diff
@@ -1486,6 +1495,13 @@ void Reducer::finalize_backward() {
     local_used_maps_reduced_ = false;
   }
 
+  // Reset bucket counting to ensure we can appropriately launch allreduce for
+  // each bucket in the next backwards.
+  reset_bucket_counting();
+  // If we populated rebuilt params list in this backward call, avoid
+  // repopulating in subsequent backward calls.
+  all_rebuilt_params_pushed_ = !rebuilt_params_.empty();
```

Comment:
rebuilt_params_() will be cleared after rebuild_bucket(), in this case, all_rebuilt_params_pushed_ will be true in every iteration after rebuilding bucket, and turns out to push rebuilt params many times, right?

Reply:
Good point, although pushing rebuilt params is also guarded by `should_rebuild_bucket()` call, which will return false after we have rebuilt buckets, so this wouldn't cause an issue from what I can tell.

Although, it's probably better to change this to:
```
all_rebuilt_params_pushed = all_rebuilt_params_pushed || !rebuilt_params_.empty()

``